aips/wallet-standard.md

```diff
@@ -0,0 +1,325 @@
+  features: AptosFeatures;
+  accounts: readonly AptosWalletAccount[];
+}
+```
+
+<ins>AptosWalletAccount interface implementation</ins>
+
+A wallet must implement a [AptosWalletAccount interface](https://github.com/aptos-labs/wallet-standard/blob/main/src/account.ts) that represents the accounts that have been authorized by the dapp.
+
+```ts
+enum AptosAccountVariant {
+  Ed25519,
+  MultiEd25519,
+  SingleKey,
+  MultiKey,
+}
+
+class AptosWalletAccount implements WalletAccount {
+  address: string;
+
+  publicKey: Uint8Array;
+
+  chains: AptosChain;
+
+  features: AptosFeatures;
+
+  variant: AptosAccountVariant;
+
+  label?: string;
+
+  icon?:
+    | `data:image/svg+xml;base64,${string}`
+    | `data:image/webp;base64,${string}`
+    | `data:image/png;base64,${string}`
+    | `data:image/gif;base64,${string}`
+    | undefined;
+}
+```
+
+<ins>Register Wallet</ins>
+
+A wallet registers itself using the [registerWallet](https://github.com/wallet-standard/wallet-standard/blob/master/packages/core/wallet/src/register.ts#L25) method to notify the dapp it is ready to be registered.
+
+```ts
+const myWallet = new MyWallet();
+
+registerWallet(myWallet);
+```
+
+**Dapp**
+
+<ins>Get Wallets</ins>
+
+A dapp uses the [getAptosWallets()](https://github.com/aptos-labs/wallet-standard/blob/main/src/detect.ts#L30) function which gets all the Aptos standard compatible wallets.
+
+```ts
+import { getAptosWallets } from "@aptos-labs/wallet-standard";
+
+let { aptosWallets, on } = getAptosWallets();
+```
+
+<ins>Register Events</ins>
+
+On first load, and before the dapp has been loaded, it gets all the wallets that have been registered so far. To keep getting all the registered wallets after this point, the dapp must add an event listener for new wallets that get registered receiving an unsubscribe function, which it can later use to remove the listener.
+
+```ts
+const removeRegisterListener = on("register", function () {
+  // The dapp can add new aptos wallets to its own state context as they are registered
+  let { aptosWallets } = getAptosWallets();
+});
+
+const removeUnregisterListener = on("unregister", function () {
+  let { aptosWallets } = getAptosWallets();
+});
+```
+
+The dapp has an event listener now, so it sees new wallets immediately and doesn't need to poll or list them again.
+This also works if the dapp loads before any wallets (it will initialize, see no wallets, then see wallets as they load)
+
+<ins>Wallet Request</ins>
+
+A dapp makes a wallet request by calling the feature name that coresponds to the desired action.
+
+```ts
+const onConnect = () => {
+  this.wallet.features["aptos:connect"].connect();
+};
+```
+
+## Risks and Drawbacks
+
+The new standard uses the [new TypeScript SDK](https://github.com/aptos-labs/aptos-ts-sdk) types and therefore requires dapps and wallets to use/migrate to the new TypeScript SDK or hold a conversion layer from the new TypeScript SDK types to the legacy TypeScript SDK types.
```

Comment:
Just throwing this out there, but it'd be really cool to have a migration tool, that automatically converts your code to use the new SDK.

I know it's not trivial though.

Reply:
agreed. the adapter (and I believe also Petra plugin) already uses this kind of a conversion layer, we can get it into its own package to be used by wallets